src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -10,143 +13,335 @@ module Dialogs
-              @max_size.to_human_string
-            )
-          )
+          w = current_widget
+          w.store
+          @ptemplate.region = w.region
+          @ptemplate.size_choice = value
         end
+      end
 
-        def manual_size_term
-          # radio button text
-          term(
-            :LeftRadioButtonWithAttachment,
-            Id(:manual_size),
-            Opt(:notify),
-            _("Custom Size"),
-            VBox(
-              Id(:manual_size_attachment),
-              MinWidth(
-                15,
-                InputField(Id(:size_input), Opt(:shrinkable), _("Size"))
-              )
-            )
-          )
+      # An invisible widget that knows a Region
+      class MaxSizeDummy < CWM::Empty
+        attr_reader :region
+
+        def initialize(region)
+          @region = region
         end
 
-        def manual_region_term
-          min_block = @disk.region.start
-          # FIXME: libyui widget overflow :-(
-          max_block = min_block + @disk.region.length - 1
-          start_block = @slot.region.start
-          end_block = start_block + @slot.region.length - 1
-
-          # radio button text
-          term(
-            :LeftRadioButtonWithAttachment,
-            Id(:manual_region),
-            Opt(:notify),
-            _("Custom Region"),
-            VBox(
-              Id(:manual_region_attachment),
-              MinWidth(
-                10,
-                IntField(
-                  Id(:start_block),
-                  _("Start Block"),
-                  min_block, max_block, start_block
-                )
-              ),
-              MinWidth(
-                10,
-                IntField(
-                  Id(:end_block),
-                  _("End Block"),
-                  min_block, max_block, end_block
-                )
+        # @macro seeAbstractWidget
+        def store
+          # nothing to do, that's OK
+        end
+      end
+
+      # Enter a human readable size
+      class CustomSizeInput < CWM::InputField
+        # @return [Y2Storage::DiskSize]
+        attr_reader :min_size, :max_size
+
+        def initialize(ptemplate, regions)
+          textdomain "storage"
+          @ptemplate = ptemplate
+          @regions = regions
+          largest_region = @regions.max_by(&:size)
+          @max_size = largest_region.size
+          @min_size = Y2Storage::DiskSize.new(1)
+        end
+
+        # Forward to ptemplate
+        def size
+          @ptemplate.custom_size
+        end
+
+        # Forward to ptemplate
+        def size=(v)
+          @ptemplate.custom_size = v
+        end
+
+        # @return [Y2Storage::Region] the smallest region
+        #   that can contain the chosen size
+        def parent_region
+          suitable_rs = @regions.find_all { |r| r.size >= size }
+          suitable_rs.min_by(&:size)
+        end
+
+        # @return [Y2Storage::Region] create it in the smallest region
```

Comment:
it return it should be like `# @return [Y2Storage::Region] newly created region that is in the smallest region that can contain it`. or place this `create it` outside of return statement and have something liek
```
# creates region in the smallest free region where it fits
# @return [Y2Storage::Region]
```